sdk/keyvault/azcertificates/models.go

```diff
@@ -590,7 +590,7 @@ func issuerParametersFromGenerated(g *generated.IssuerParameters) *IssuerParamet
 // KeyProperties - Properties of the key pair backing a certificate.
 type KeyProperties struct {
 	// Elliptic curve name. For valid values, see JsonWebKeyCurveName.
-	Curve *JSONWebKeyCurveName `json:"crv,omitempty"`
+	Curve *CertificateKeyCurveName `json:"crv,omitempty"`
```

Comment:
We don't expose anything like `KeyProperties` for certificates. All this (and with slightly different names, like `KeyCurveName` here) hangs off of `CertificatePolicy`, which is more or less the input to create a certificate: https://github.com/Azure/azure-sdk-for-net/blob/75b74e3d5bdd65a7d34c5087faded65595b6a23a/sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificatePolicy.cs#L163-L183

Or, if it's easier, see our APIView. This was reviewed across languages as well: https://apiview.dev/Assemblies/Review/da210106190946c0874b30d5f84f41b1

Reply:
I believe I am more inline with this API while staying idiomatic to Go. Can you take another look at this PR now?